src/routes/api/act.ts

```diff
@@ -55,25 +61,166 @@ router.get("/TWCodeCheck", async (req, res) => {
 });
 
 /**
- * [GET] /api/v1/act/new - Account creation endpoint
+ * [GET] /api/v1/account/createAccount - Account creation endpoint
  * 
  * Endpoint to create a Club Rosé account
  */
-router.put("/new", (req, res) => {
-    res.send({ success: 1 }).contentType("application/json");
+router.post("/createAccount", async (req, res) => {
+    // zod schema for the body
+    // TODO: Figure out how I can put this on the body
+    const bodySchema = z.object({
+        email: z.string().email({ message: "Invalid email address!" }),
+        passwd: z.string().min(8, { message: "Passwords cannot be shorter than 8 characters!" }).max(128, { message: "Password cannot be over 128 characters!" }),
+        miiName: z.string().min(1, { message: "Mii nicknames cannot be shorter than 1 character!" }).max(10, { message: "Mii nicknames cannot be longer than 10 characters!" }),
+        pid: z.number().min(10, { message: "Principal IDs cannot be shorter than 10 numbers!" }).max(10, { message: "Principal IDs cannot be longer than 10 numbers!" }),
+        pnid: z.string().min(6, { message: "Pretendo Network IDs cannot be shorted than 6 characters!" }).max(16, { message: "Pretendo Network IDs cannot be longer than 16 characters!" }),
+    });
+
+    // gets the required data from the html form
+    const email = req.body.email; // User's email
+    const passwd = req.body.password; // Users plaintext password (Will be hashed)
+    const miiName = req.body.nickname; // User's Mii name/Mii nickname
+    const pid = req.body.principal_id; // User's principal id
+    const pnid = req.body.id; // The users Pretendo Network ID
+    const time = dayjs().format("HH:mm:ss"); // 24 hr time
+
+    if (!email || !passwd || !miiName || !pid || !pnid) {
+        const missingParams = ["email", "passwd", "miiName", "pid", "pnid"].filter(param => !eval(param));
+
+        const missingMessage = missingParams.join(", ").replace(/, ([^,]*)$/, " and $1");
+
+        res.status(400).json({
+            endpoint: "createAccount",
+            time: time,
+            error: 1,
+            result: {
+                message: `Missing required parameters: ${missingMessage}.`,
+            },
+        });
+        return;
+    }
+
+    // hashes a the password
+    const encryptPassword = async (pass: string) => {
+        const saltRounds = 10;
+        return await bcrypt.hash(pass, saltRounds);
+    };
+
+    // gets a json web token for that account
+    const getAccountJwt = (user: object) => {
+        return jwt.sign({ user }, JWT_SECRET, { expiresIn: "1h" });
+    };
+    
+    try {
+        const hashedPasswd = await encryptPassword(passwd);
+        const newAccount = await createAccount(email, hashedPasswd, miiName, pid, pnid);
+        const jwtToken = getAccountJwt(newAccount); // json web token token lol
```

Comment:
"lol?"

Reply:
because jwt stands for json web token and the variable is jsonwebtokenToken, idk i found it funny